docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,594 @@
+
+.. code::
+
+    - dataset:
+        type: spectrum-dataset
+        maps:
+            counts: "obs-123/counts.fits"
+            exposure: "obs-123/exposure.fits"
+            edisp: "obs-123/edisp.fits"
+        model: "model.yaml"
+        background-model: "obs-123/background.fits"
+        likelihood: "wstat"
+
+    - dataset:
+        type: spectrum-dataset
+        maps:
+            counts: "obs-124/counts.fits"
+            exposure: "obs-124/exposure.fits"
+            edisp: "obs-124/edisp.fits"
+        model: "model.yaml"
+        background-model: "obs-124/background.fits"
+        likelihood: "wstat"
+
+    - dataset:
+        type: flux-point-dataset
+        flux-points : "fermipy-flux-points.fits"
+        model: "spectral-model.yaml"
+        likelihood: "template"
+
+
+Task List
+=========
+
+This is a proposal for a list of pull requests implementing the proposed changes,
+ordered by priority:
+
+1. Refactor the current `FluxPointFit` into a `FluxPointsDataset` class
+and make it work with the current `Fit` class. Ensure that the `Fit` class
+supports the old inheritance scheme as well as the new dataset on init.
+Update tests and tutorials (`#2023 <https://github.com/gammapy/gammapy/pull/2023>`_).
+
+2. Refactor the current `MapFit` into the `MapDataset` class. Only support a
+fixed energy dispersion and psf first. Use the current `MapEvaluator` for model
+evaluation, but split out the background evaluation. Update test and tutorials
+(`#2026 <https://github.com/gammapy/gammapy/issues/2026>`_)
+
+3. Implement the `Datasets` class in `gammapy.utils.fitting.datasets`, add tests
+using multiple `MapDataset`. Change the `Fit` interface to take a `Datasets`
+object or list of `MapDataset` on input (`#2030 <https://github.com/gammapy/gammapy/issues/2030>`_).
+
+4. Enable joint-likelihood analyses by filtering the `Datasets.parameters` for
+unique parameters only. Add a tutorial for joint-likelihood fitting of multiple
+observations (`#2045 <https://github.com/gammapy/gammapy/issues/2045>`_).
+
+5. Implement `MapDataset.setup()` method, using a list of `MapEvaluator`
+objects. Add an `.evaluation_radius()` attribute to all the spatial models.
+Only support a fixed PSF and edisp per model component (`#2071 <https://github.com/gammapy/gammapy/issues/2071>`_).
+
+6. Add support for psf maps to `MapDataset.setup()`. Extend the `.setup()` method
+to look up the correct PSF for the given model component. A `PSFMap` has to be
+passed on `MapDataset.__init__()` (**v0.14**).
+
+7. Add support for energy dispersion maps to `MapDataset`. Extend the `.setup()`
+method to look up the correct EDisp for a given model component. An `EdispMap`
+has to be passed on `MapDataset.__init__()` (**v0.14**).
+
+8. Add tutorial for joint-likelihood fitting of IACT and Fermi-LAT data, based
+on the joint-crab dataset (**v0.14**).
+
+9. Add a `name` attribute to datasets and a `Datasets.__getitem__` method (**v0.14**).
```

Comment:
Did you discuss introducing names for datasets?
I think as for models, that's something we might or might not want.
If you really propose to add it and mention it in the task list here, you should probably say something int he PIG how it would work. Do names have to be unique? Who generates them, in what way? Do we in the framework ever rely on and access by name? Or is this just something for user display, e.g. an obs_id?
One use case could be the lightcurve analysis, where you either have one run split into few-minute chunks, or you have time bins that span many full and partial observations. Is it the responsibility of the Gammapy framework to generate names there? Does reading back datasets for lightcurve analysis rely on those names?

My naive feeling is that we want to avoid names, or if you allow them, then they will be "user-land", never used within Gammapy. Probably a `Dataset` should have a `meta` dict as user-land? That might be something you could add in the PIG?